osquery/core/sql.cpp

```diff
@@ -69,12 +69,23 @@ SQL::SQL(const std::string& q) {
   status_ = Status(code, getStringForSQLiteReturnCode(code));
 }
 
-QueryData SQL::rows() { return results_; }
+const QueryData& SQL::rows() { return results_; }
 
 bool SQL::ok() { return status_.ok(); }
 
+Status SQL::getStatus() { return status_; }
+
 std::string SQL::getMessageString() { return status_.toString(); }
 
+const std::string SQL::kHostColumnName = "_source_host";
```

Comment:
I've thought about adding "special" columns prefixed with "$", "@", or "#". An underscore works, but we could reserve "$" for runtime or querytime generated columns such as a distributed query or other trigger.

Another big need is when you want everything matching an FS regex like path = "*/bin/*". Right now the table will generate the correct information but SQL will filter it out since none of the return paths match the SQL EQUALS/MATCHES operator. But if we did $path =, then we could set that trigger to the regex and emit the normal path as the unique-match. And we cannot expose the infix (like, glob, regex) or OR to virtual table logic.